pyannotate_runtime/tests/test_collect_types.py

```diff
@@ -594,6 +594,43 @@ def func_with_unknown_module_types(c):
 
         self.assert_type_comments('func_with_unknown_module_types', ['(C) -> C'])
 
+    def test_yield_basic(self):
+        # type: () -> None
+        def gen(n, a):
+            for i in range(n):
+                yield a
+
+        with self.collecting_types():
+            list(gen(10, 'x'))
+
+        self.assert_type_comments('gen', ['(int, str) -> Iterator[str]'])
+
+    def test_yield_various(self):
+        # type: () -> None
+        def gen(n, a, b):
+            for i in range(n):
+                yield a
+                yield b
+
+        with self.collecting_types():
+            list(gen(10, 'x', 1))
+            list(gen(0, 0, 0))
+
+        # TODO: Wish this could return Iterator[Union[int, str]]
```

Comment:
I'd argue this **must** return `Iterator[Union[int, str]]`. In the current form any iterator produces more then 1 concrete type in the same run would be collected incorrectly. This is probably an uncommon scenario, so it's fine to leave it as is, but it's worth to add comments about it in both tests and implementation.

Reply:
I agree. But I also want to release 1.0.4 before hack week. Maybe I'll do it this way for now and iterate later.